backend/src/main/java/codezap/template/service/TemplateService.java

```diff
@@ -136,27 +135,39 @@ public FindAllTemplatesResponse findAllBy(
         pageable = PageRequest.of(pageable.getPageNumber() - 1, pageable.getPageSize(), pageable.getSort());
 
         if (categoryId != null && tagIds != null) {
-            List<Long> templateIds = templateTagRepository.findAllTemplateIdInTagIds(tagIds, tagIds.size());
+            List<Long> templateIds = findTemplateIdContainsTagIds(tagIds);
+            categoryRepository.fetchById(categoryId);
```

Comment:
이 부분이 카테고리의 존재 유무를 확인하기 위한 메서드 호출이 맞을까요?
`fetchById()` 메서드의 반환값을 받지 않고 있어서요!
그렇다면 상황에 더 맞는 메서드를 사용해 주는 것이 가독성에 좋을 것 같습니다..!
```suggestion
            categoryRepository.existsById(categoryId);
```

Reply:
어머 해당 부분에 대해서는 인지 못하고 있었네요..!

fetchById를 쓴 이유는 fetchById에서 존재하지 않을 경우 예외처리를 해줘서입니다. 따로 existsById를 쓰면 동일한 예외 처리 로직이 service의 중복 처리 될 것 같네요 ..! 
근데 이건 선호도 문제일 것 같아요 ! 혹시 짱수의 선호도를 알 수 있을까여 ? 
1. fetchById를 사용해서 예외처리를 한 곳에서 처리
2. existsById를 사용해서 상황에 맞는 적절한 코드 사용

> 저도 적고 보니 2번이 맞는 것 같네요 ㅎㅎ

Reply:
적고 보니 2번이 맞는 것 같아서 고칠게요 ~